### DISCUSSION TITLE
Linux does not appear as a target OS in Visual Studio Avalonia project

### DISCUSSION THREAD

**Discussion by cakkasli** [AvaloniaUI/Avalonia#7796] [Q&A]
I use Windows 10 with Visual Studio 2022; and want to make a GUI application targeting Linux using Avalonia.

I followed all the instructions but under target OS Linux does not show up as in my screen shot:

![image](https://user-images.githubusercontent.com/81676515/158062509-9c1bb46f-f078-4c19-87c0-910916dd1c3d.png)

Could someone help me with this?

**Comment by jp2masa** [comment]
I think (None) should work on Windows, macOS and Linux. You should only select a different option if you need specific runtime APIs for one of those OSes, or for Android/iOS/tvOS, which use Xamarin and where you can't run a .NET executable like on desktop platforms (i.e. need AOT compilation or additional build steps).

**Reply by cakkasli** [reply]
But there is no elf file under \AvaloniaApplication2\bin\Debug\net6.0\runtimes\linux-x64\native How should I run it under Linux?

**Reply by jp2masa** [reply]
The idea is that you would run the app with `dotnet AvaloniaApplication2.dll`, even with normal publish (i.e. [cross-platform binary](https://docs.microsoft.com/en-us/dotnet/core/deploying/#produce-a-cross-platform-binary)). The alternatives are [framework-dependent](https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-framework-dependent) and [self-contained](https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained), both of which should create the ELF file you're expecting.
